### DISCUSSION TITLE
forking-renovate external-host-error: 404 on PATCHing main branch

### DISCUSSION THREAD

**Discussion by simonbasle** [renovatebot/renovate#13800] [Request Help]
Since at least a couple of days, the forking-renovate GitHub app fails with a 404 while trying to fast-forward the fork's main branch using a PATCH request.

See logs below:

```
WARN: Error updating fork from upstream - cannot continue
{
  "url": "repos/renovate-bot/reactor-core/git/refs/heads/main",
  "sha": "4081f49202772f6f488ea14ee9e231cf77645eed",
  "err": {
    "name": "HTTPError",
    "code": "ERR_NON_2XX_3XX_RESPONSE",
    "timings": {
      "start": 1643109640205,
      "socket": 1643109640205,
      "lookup": 1643109640206,
      "connect": 1643109640215,
      "secureConnect": 1643109640227,
      "upload": 1643109640227,
      "response": 1643109640432,
      "end": 1643109640432,
      "phases": {
        "wait": 0,
        "dns": 1,
        "tcp": 9,
        "tls": 12,
        "request": 0,
        "firstByte": 205,
        "download": 0,
        "total": 227
      }
    },
    "message": "Response code 404 (Not Found)",
    "stack": "HTTPError: Response code 404 (Not Found)\n    at Request.<anonymous> (/home/ubuntu/renovateapp/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
    "options": {
      "headers": {
        "user-agent": "Renovate Bot (GitHub App 2740)",
        "accept": "application/vnd.github.v3+json",
        "authorization": "***********",
        "content-type": "application/json",
        "content-length": "63",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://api.github.com/repos/renovate-bot/reactor-core/git/refs/heads/main",
      "hostType": "github",
      "username": "",
      "password": "",
      "method": "PATCH",
      "http2": false
    },
```

I should have noticed no PR from Renovate for a few days while dependabot was updating some common dependencies in another project I watch...

I actually noticed the app dashboard seems to work when using a direct link to the forked repo, and that's where I saw all available logs (past 3 days) display that `external-host-error` status.

See detail of latest job, triggered by checking the box in the Dependency Dashboard issue, at https://app.renovatebot.com/dashboard#github/reactor/reactor-core/562486946

**Comment by rarkins** [comment]
It's a bug on GitHub's side which comes and goes. I reported it to them but until now I've fixed up each repo manually before they have the time to investigate.

Seeing as this PATCH command now seems to intermittently fail, I'm implementing #13586 as a long term solution.

**Reply by rarkins** [reply]
Good news: it seems to be working. `reactor/reactor-core` just processed fine with the new code, and here's the relevant new log:

```
{"level":20,"msg":"Resetting main to upstream","time":"2022-01-25T20:11:07.265Z"}
{"level":20,"msg":"resetToBranch(main)","time":"2022-01-25T20:11:09.415Z"}
{"level":20,"resetLog":"HEAD is now at f5dae366 Rewrite a MonoSubscribeOn test to be less flaky (#2893)\n","msg":"git reset log","time":"2022-01-25T20:11:09.964Z"}
{"level":20,"pushLog":{"pushed":[],"update":{"head":{"local":"refs/heads/main","remote":"refs/heads/main"},"hash":{"from":"2835f17d","to":"f5dae366"}},"repo":"https://github.com/renovate-bot/reactor-core.git","ref":{"local":"refs/remotes/origin/main"},"remoteMessages":{"all":[]}},"msg":"git push log","time":"2022-01-25T20:11:10.855Z"}
```

**Reply by simonbasle** [reply]
thanks for the quick turnaround!
it is still blocked, though, complaining about a configuration issue despite the fact I've since fixed in the upstream repo main branch.
could it be that the bot reads the configuration from the origin/forked repo before it has fetched the latest changes from upstream? 🤔

**Reply by simonbasle** [reply]
seems to be working again as of 17 hours ago (wed. 4:50pm CET) 🤷
